conanfile.py

```diff
@@ -0,0 +1,364 @@
+            match_label = re.match(r'''^\s*"(node[0-9]+)"\s*\[\s*label\s*=\s*"([^\\"]+)''', row)
+            if match_label:
+                node = match_label.group(1)
+                label = match_label.group(2)
+                if label.startswith("LLVM"):
+                    yield node, f"llvm-core::{label}"
+                # XXX find_library adds direct filepath -> imho a flaw in current cmake files
+                elif label.endswith("libsqlite3.a"):
+                    yield node, "sqlite3::sqlite3"
+                elif label.endswith("libclang-cpp.so"):
+                    yield node, "clang::clang"
+                elif label.endswith("libclangCodeGen.a"):
+                    yield node, "clang::clangCodeGen"
+                elif label.endswith("libclangTooling.a"):
+                    yield node, "clang::clangTooling"
+                else:
+                    yield node, label_replacements.get(label, label)
+
+    def node_dependencies(dot):
+        ignore_deps = [
+        ]
+        labels = {k: v for k, v in node_labels(dot)}
+        for row in dot:
+            # "node0" -> "node1" [ style = dashed ] // phasar -> LLVMAnalysis
+            match_dep = re.match(r'''^\s*"(node[0-9]+)"\s*->\s*"(node[0-9]+)".*''', row)
+            if match_dep:
+                node_label = labels[match_dep.group(1)]
+                dependency = labels[match_dep.group(2)]
+                if node_label.startswith("phasar") and PurePosixPath(dependency).parts[-1] not in ignore_deps:
+                    yield node_label, labels[match_dep.group(2)]
+        # some components don't have dependencies
+        for label in labels.values():
+            if label.startswith("phasar"):
+                yield label, None
+
+    system_libs = {
+        "ole32",
+        "delayimp",
+        "shell32",
+        "advapi32",
+        "-delayload:shell32.dll",
+        "uuid",
+        "psapi",
+        "-delayload:ole32.dll",
+        "ntdll",
+        "ws2_32",
+        "rt",
+        "m",
+        "dl",
+        "pthread",
+        "stdc++fs"
+    }
+    components = {}
+    dotfile_rows = dotfile.split("\n")
+    for node, dependency in node_dependencies(dotfile_rows):
+        if dependency in system_libs:
+            key = "system_libs"
+        elif dependency is not None and (dependency.startswith("phasar") or "::" in dependency):
+            key = "requires"
+        else:
+            key = "unknown"
+        if node not in components:
+            components[node] = { "system_libs": [], "requires": [], "unknown": [] }
+            if dependency is not None:
+                components[node][key] = [dependency]
+        elif dependency is not None:
+            components[node][key].append(dependency)
+
+    return components
+
+class PhasarRecipe(ConanFile):
+    name = "phasar"
+    package_type = "library"
+
+    # Optional metadata
+    license = "MIT license"
+    author = "<Put your name here> <And your email here>"
```

Comment:
Are these TODOs?

Reply:
is optional and repository url is already provided, therefore removed https://docs.conan.io/2/reference/conanfile/attributes.html#author